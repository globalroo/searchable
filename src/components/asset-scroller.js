import React from "react";

import "./shared.css";

import { AssetTile } from "./asset-tile";
import { GridList } from "@material-ui/core";
import { MEDIA_TYPE } from "../tmdb-service/tmdb-api";

const styles = {
	root: {
		display: "flex",
		flexWrap: "wrap",
		justifyContent: "space-between",
		overflow: "hidden"
	},
	gridList: {
		flexWrap: "nowrap",
		transform: "translateZ(0)",
		paddingTop: 3,
		paddingBottom: 3
	}
};

export const AssetScroller = ({ assets, media_type = MEDIA_TYPE.MOVIE }) => {
	return (
		<div style={styles.root}>
			<GridList data-testid="asset-grid" style={styles.gridList}>
				{assets.map((asset, ix) => {
					const mappedAsset = asset.media_type ? asset : { ...asset, media_type };
					const key = `${asset.id}_${asset.media_type}_${ix}`;
					return <AssetTile asset={mappedAsset} key={key} data-testid={key} />;
				})}
			</GridList>
		</div>
	);
};
