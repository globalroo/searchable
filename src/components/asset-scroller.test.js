import React from "react";
import { render, cleanup } from "react-testing-library";
import { AssetScroller } from "./asset-scroller";

import mockMovie from "src/test-helpers/mock-movie";

describe("Asset Scroller tests", () => {
	afterEach(cleanup);
	it("Renders an AssetScroller component", () => {
		const testData = { ...mockMovie, media_type: "Defined" };
		const { getByTestId } = render(<AssetScroller assets={[testData]} />);
		const assetGrid = getByTestId("asset-grid");
		expect(assetGrid).toMatchSnapshot();
	});
	it("Renders an AssetScroller and overrides the media type if specified", () => {
		const testData = { ...mockMovie };
		const { getByTestId } = render(<AssetScroller assets={[testData]} media_type="person" />);
		const assetGrid = getByTestId("asset-grid");
		expect(assetGrid).toMatchSnapshot();
	});
});
