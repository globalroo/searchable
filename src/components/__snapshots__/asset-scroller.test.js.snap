// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asset Scroller tests Renders an AssetScroller and overrides the media type if specified 1`] = `
<ul
  class="MuiGridList-root-17"
  data-testid="asset-grid"
  style="margin: -2px; flex-wrap: nowrap; transform: translateZ(0); padding-top: 3px; padding-bottom: 3px;"
>
  <div
    class="MuiGridListTile-root-18 pointer-hover image-tile"
    data-testid="226486_Tammy"
    style="padding: 5px; height: 225px; width: 150px;"
  >
    <div
      class="MuiGridListTile-tile-19"
    >
      <img
        alt="Tammy"
        src="http://image.tmdb.org/t/p/w300//cq6wvDqETqJXgpQplkL0FBw2leM.jpg"
      />
      <div
        class="MuiGridListTileBar-root-22 MuiGridListTileBar-titlePositionBottom-23"
      >
        <div
          class="MuiGridListTileBar-titleWrap-26"
        >
          <div
            class="MuiGridListTileBar-title-29"
          >
            Tammy
          </div>
        </div>
      </div>
    </div>
  </div>
</ul>
`;

exports[`Asset Scroller tests Renders an AssetScroller component 1`] = `
<ul
  class="MuiGridList-root-1"
  data-testid="asset-grid"
  style="margin: -2px; flex-wrap: nowrap; transform: translateZ(0); padding-top: 3px; padding-bottom: 3px;"
>
  <div
    class="MuiGridListTile-root-2 pointer-hover image-tile"
    data-testid="226486_Tammy"
    style="padding: 5px; height: 225px; width: 150px;"
  >
    <div
      class="MuiGridListTile-tile-3"
    >
      <img
        alt="Tammy"
        src="http://image.tmdb.org/t/p/w300//cq6wvDqETqJXgpQplkL0FBw2leM.jpg"
      />
      <div
        class="MuiGridListTileBar-root-6 MuiGridListTileBar-titlePositionBottom-7"
      >
        <div
          class="MuiGridListTileBar-titleWrap-10"
        >
          <div
            class="MuiGridListTileBar-title-13"
          >
            Tammy
          </div>
        </div>
      </div>
    </div>
  </div>
</ul>
`;
